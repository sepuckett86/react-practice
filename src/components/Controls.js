import React from 'react';
import PropTypes from 'prop-types';

function Controls({ handleChange, controls }) {
  return (
    <>
    <label>
      Title: <input name="title" type="text" onChange={handleChange} value={controls.title}></input>
    </label>
    <label>
      Subtitle: <input name="subtitle" type="text" onChange={handleChange} value={controls.subtitle}></input>
    </label>
    </>
  );
}

Controls.propTypes = {
  handleChange: PropTypes.func.isRequired,
  controls: PropTypes.shape({
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
  }).isRequired
};

export default Controls;
