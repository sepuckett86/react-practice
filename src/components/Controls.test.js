import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls';

describe('Controls component', () => {
  it('renders Controls', () => {
    const wrapper = shallow(
      <Controls 
        controls={{ title: '', subtitle: '' }}
        handleChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
