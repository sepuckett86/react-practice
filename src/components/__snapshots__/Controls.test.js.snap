// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders Controls 1`] = `
<Fragment>
  <label>
    Title: 
    <input
      name="title"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label>
    Subtitle: 
    <input
      name="subtitle"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
</Fragment>
`;
